src/benchmarklib/table_builder.hpp

```diff
@@ -0,0 +1,241 @@
+#pragma once
+
+#include <optional>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "boost/hana/assert.hpp"
+#include "boost/hana/for_each.hpp"
+#include "boost/hana/tuple.hpp"
+#include "boost/hana/zip_with.hpp"
+
+#include "resolve_type.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+// helper structs and functions for compile time, the actual code is in namespace opossum
+namespace table_builder {
+// similar to std::optional but has_value is known at compile time, so "if constexpr" can be used
```

Comment:
Did you write this code or is it based on a StackOverflow post (or alike)? If so, please add a comment.

Reply:
That's my code.

Reply:
sweet

Reply:
As in "cool, I like it"